src/mame/drivers/raiden_ms.cpp

```diff
@@ -374,9 +384,10 @@ void raiden_ms_state::audio_map(address_map &map)
 	map(0xd000, 0xd7ff).ram();
 	// area 0xdff0-5 is never ever readback, applying a RAM mirror causes sound to go significantly worse,
 	// what they are even for?  (offset select bankswitch rather than data select?)
-	map(0xdff0, 0xdfff).r(m_soundlatch, FUNC(generic_latch_8_device::read)).w(FUNC(raiden_ms_state::unk_snd_dffx_w));
-	map(0xe000, 0xe001).w(m_ym1, FUNC(ym2203_device::write));
-	map(0xe002, 0xe003).w(m_ym2, FUNC(ym2203_device::write));
+	map(0xdff0, 0xdff7).w(FUNC(raiden_ms_state::unk_snd_dffx_w));
+	map(0xdff8, 0xdff8).rw(m_soundlatch[1], FUNC(generic_latch_8_device::read), FUNC(generic_latch_8_device::clear_w));
+	map(0xdff9, 0xdff9).r(m_soundlatch[0], FUNC(generic_latch_8_device::read));
```

Comment:
As written, this will clear the flag for `m_soundlatch[1]` on a read or write, but only clear the flag for `m_soundlatch[0]` on a read.  If it clears one of them on either a read or write, wouldn’t it be more likely that it just doesn’t decode the R/W line for the purpose of clearing the flag on accesses to those addresses?  I wouldn’t think it would decode R/W for one but not for the other.

Reply:
I'm not sure what you mean about clearing on reads, since reading either latch shouldn't be clearing anything at all. The sound CPU just writes to $dff8 to clear/acknowledge the latched command after reading/handling both addresses individually.

Reply:
It is more likely that the $dff8 write clears both latches then.

Reply:
Clearing them both works, since the audio CPU only reads $dff9 if $dff8 is also not zero, and the main CPU always writes both at once.

Reply:
If you don’t want it to clear on reads, you need to `.set_separate_acknowledge(true)` in your machine configuration.  By default, the generic latch devices clear the flag on read.

Reply:
I'm talking about clearing the _value_, not `m_latch_written`. The latter doesn't even affect the driver at all, unless I'm missing something.

Reply:
Ah right.  So it treats a zero in the latch(es) as no command pending?

Reply:
Right, if $dff8 is anything other than zero, it reads both latches as a command/parameter pair. After that, it writes to $dff8 to clear the pending command until the main CPU writes another one.

($dff9 isn't read at all _unless_ $dff8 is also non-zero, which is why I was originally only clearing one latch and not both. It shouldn't really make a difference, though.)